src/main/java/hudson/plugins/sonar/utils/SonarUtils.java

```diff
@@ -66,12 +75,47 @@ private SonarUtils() {
    * Read logs of the build to find URL of the project dashboard in Sonar
    */
   public static String extractSonarProjectURLFromLogs(Run<?, ?> build) throws IOException {
+    return extractPatternFromLogs(URL_PATTERN_IN_LOGS, build);
+  }
+
+  public static <T extends Action> List<T> getPersistentActions(Actionable actionable, Class<T> type) {
+    List<T> filtered = new LinkedList<T>();
+
+    // we use this method to avoid recursively calling transitive action factories
+    for (Action a : actionable.getActions()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "getActions"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)